submissions/TaniaBondarenko/friends-app/app.js

```diff
@@ -0,0 +1,130 @@
+
+function refreshPage() {
+        const wholePage = document.querySelector("body");
+        wholePage.setAttribute("class", "refresh");
+        wholePage.innerHTML = `<div>Ops...Something went wrong.</div>
+        <div>Refresh the page to start finding friends</div>
+        <button class="refresh_button">Click to refresh</button>` ;
+        document.querySelector(".refresh_button").addEventListener("click", fetchFriend);
+}
+
+ function addFriends(friendsToBeAdded) {
+      let fragment = document.createDocumentFragment();
+        friendsToBeAdded.forEach(friend=> {
+                let friendCard = document.createElement("div");
+               friendCard.setAttribute("class", "friend_card");
+                friendCard.innerHTML = `<div class="card_wrapper ${friend.gender}"><div class="name ">${`${friend.name.first} ${friend.name.last}`}</div>
+                                <div class="photo"><img src="${friend.picture.large}"></div>
+                                <div class="info_block">
+                                <div class="age">Age ${friend.dob.age}</div>
+                                <div class="place">${friend.location.city}</div>
+                                <div class="email"><a href="mailto:${friend.mail}" class="email_link">${friend.email}</a></div>
+                                <div class="gender">${friend.gender}</div>
+                                </div></div>
+                                `;
+                fragment.appendChild(friendCard);       
+        });
+              friends.appendChild(fragment);
+
+};
+
+document.querySelector(".sidebar").addEventListener("click", sortFriends);
+
+function sortFriends({ target }) {
+        chooseAppropriateFriends();
+        switch (target.className||target.id) {
+                case "ascendent_by_name":
+                        arrForFilter.sort((a, b) => a.name.first.localeCompare(b.name.first));
+                          redrawfriends(); 
+                        break;
+                case "descendent_by_name":
+                        arrForFilter.sort((a, b) => b.name.first.localeCompare(a.name.first));
+                          redrawfriends(); 
+                        break;
+                case "ascendent_by_age":
+                        arrForFilter.sort((a, b) => a.dob.age - (b.dob.age));
+                          redrawfriends(); 
+                        break;
+                case "descendent_by_age":
+                        arrForFilter.sort((a, b) => b.dob.age - (a.dob.age));
+                          redrawfriends(); 
+                        break;
+                case "male":
+                case "female":
+                case "all":
+                        letFilter();
+                        break;      
+        }
+         
+}
+
+function letFilter() {
+        makeContainerEmpty();
+        if (document.getElementById("male").checked) {
+                justMale = friendsData.filter(el => el.gender === "male");
```

Comment:
You can take the value of selected input and filter by this value

Reply:
Can I leave it as it is now?

Reply:
Ehm... no? You should re-write this and make it good :)

Try to change it.

Reply:
okay